src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.cs

```diff
@@ -426,6 +435,37 @@ private void EnqueueDocumentChangedEvent(Solution oldSolution, Solution newSolut
                         var newProject = newSolution.GetRequiredProject(documentId.ProjectId);
 
                         await EnqueueChangedDocumentWorkItemAsync(oldProject.GetRequiredDocument(documentId), newProject.GetRequiredDocument(documentId)).ConfigureAwait(false);
+
+                        if (WorkspaceConfigurationOptions.ShouldConnectSourceGeneratedFilesToWorkspace(newSolution.Options))
+                        {
+                            var oldProjectSourceGeneratedDocuments = await oldProject.GetSourceGeneratedDocumentsAsync(_shutdownToken).ConfigureAwait(false);
+                            var oldProjectSourceGeneratedDocumentsById = oldProjectSourceGeneratedDocuments.ToDictionary(static document => document.Id);
+                            var newProjectSourceGeneratedDocuments = await newProject.GetSourceGeneratedDocumentsAsync(_shutdownToken).ConfigureAwait(false);
+                            var newProjectSourceGeneratedDocumentsById = newProjectSourceGeneratedDocuments.ToDictionary(static document => document.Id);
+
+                            foreach (var (oldDocumentId, _) in oldProjectSourceGeneratedDocumentsById)
+                            {
+                                if (!newProjectSourceGeneratedDocumentsById.ContainsKey(oldDocumentId))
+                                {
+                                    // This source generated document was removed
+                                    EnqueueFullDocumentEvent(oldSolution, oldDocumentId, InvocationReasons.DocumentRemoved, "OnWorkspaceChanged");
+                                }
+                            }
+
+                            foreach (var (newDocumentId, newDocument) in newProjectSourceGeneratedDocumentsById)
+                            {
+                                if (!oldProjectSourceGeneratedDocumentsById.TryGetValue(newDocumentId, out var oldDocument))
+                                {
+                                    // This source generated document was added
+                                    EnqueueFullDocumentEvent(newSolution, newDocumentId, InvocationReasons.DocumentAdded, "OnWorkspaceChanged");
+                                }
+                                else
+                                {
+                                    // This source generated document may have changed
+                                    await EnqueueChangedDocumentWorkItemAsync(oldDocument, newDocument).ConfigureAwait(continueOnCapturedContext: false);
+                                }
```

Comment:
This is really funky how some of the Enqueue* methods are fire-and-forget and then this other one I guess is actually doing an async enqueue to *another* system. This isn't your doing at all and honestly your prior PR made this a bit clearer, but it makes me a bit nervous that we're mixing queue concepts. I guess it's probably fine since stuff will *eventually* get reprocessed.